### PR TITLE
[fix](nereids)disable parallel scan in some case

### LINKED ISSUE #24905: [Bug] WINDOW-FUNCTION-LAG not working on Doris 2.0.1
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

Git : git://VM-16-12-centos@30d35c414034a82a66ab8064de2076cea77ac38c

Version : doris-2.0.1-rc04

BuildInfo : VM-16-12-centos

BuildTime : Tue, 29 Aug 2023 10:35:46 UTC

### What's Wrong?

```sql
CREATE TABLE sequence_count_test3(
                `uid` int COMMENT 'user id',
                `date` datetime COMMENT 'date time', 
                `number` int NULL COMMENT 'number' 
                )
DUPLICATE KEY(uid) 
DISTRIBUTED BY HASH(uid) BUCKETS 1 
PROPERTIES ( 
    "replication_num" = "1"
); ```
```sql
INSERT INTO sequence_count_test3(uid, date, number) values (1, '2022-11-02 10:41:00', 1),
                                                   (2, '2022-11-02 11:41:00', 7),
                                                   (3, '2022-11-02 16:15:01', 3),
                                                   (4, '2022-11-02 19:05:04', 4),
                                                   (5, '2022-11-02 21:24:12', 5);
```

```sql
SELECT
	uid,
	DATE,
	e.NUMBER AS EVENT_GROUP,
CASE
		
		WHEN (
			(
				UNIX_TIMESTAMP( DATE ) - (
					lag ( UNIX_TIMESTAMP( DATE ), 1, 0 ) over ( PARTITION BY uid ORDER BY DATE  ) 
				) 
			) > 600 
			) THEN
			1 ELSE 0 
		END AS SESSION_FLAG 
	FROM
		sequence_count_test3 e 
```

uid | date | EVENT_GROUP | SESSION_FLAG
-- | -- | -- | --
1 | 2022-11-02 10:42:00 | 1 | 1
1 | 2022-11-02 10:41:00 | 1 | 1
2 | 2022-11-02 11:41:00 | 7 | 1
3 | 2022-11-02 16:15:01 | 3 | 1
4 | 2022-11-02 19:05:04 | 4 | 1
5 | 2022-11-02 21:24:12 | 5 | 1
1 | 2022-11-02 10:43:00 | 7 | 1

### What You Expected?

uid | date | EVENT_GROUP | SESSION_FLAG
-- | -- | -- | --
5 | 2022-11-02 21:24:12 | 5 | 1
4 | 2022-11-02 19:05:04 | 4 | 1
2 | 2022-11-02 11:41:00 | 7 | 1
1 | 2022-11-02 10:41:00 | 1 | 1
1 | 2022-11-02 10:42:00 | 1 | 0
1 | 2022-11-02 10:43:00 | 7 | 0
3 | 2022-11-02 16:15:01 | 3 | 1

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by starocean999**
## Proposed changes
close https://github.com/apache/doris/issues/24905
disable parallel scan for window function node, partition topn node and set operation node

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by starocean999** [comment]
run buildall

**Comment by starocean999** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java [hunk 1]
```diff
@@ -1739,6 +1745,13 @@ public PlanFragment visitPhysicalSetOperation(
             setPlanRoot(setOperationFragment, setOperationNode, setOperation);
         }
 
+        // in pipeline engine, we use parallel scan by default, but it broke the rule of data distribution
+        // we need turn of parallel scan to ensure to get correct result.
+        // TODO: nereids forbid all parallel scan under PhysicalSetOperation temporary
+        if (findOlapScanNodesByPassExchangeAndJoinNode(setOperationFragment.getPlanRoot())) {
+            setOperationFragment.setHasColocatePlanNode(true);
+        }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/glue/translator/PhysicalPlanTranslator.java:1753]: check set operation output physical properties, if it is any, do not set this flag

**Comment by starocean999** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**